server/controllers/stock/inventory/core.js

```diff
@@ -38,11 +39,58 @@ var errors = {
 exports.getIds = getIds;
 exports.getItemsMetadata = getItemsMetadata;
 exports.getItemsMetadataById = getItemsMetadataById;
+exports.createItemsMetadata = createItemsMetadata;
+exports.updateItemsMetadata = updateItemsMetadata;
 exports.hasBoth = hasBoth;
 exports.errors = errors;
 exports.errorHandler = errorHandler;
 
 /**
+* Create inventory metadata in the database
+*
+* @function createItemsMetadata
+* @return {Promise} Returns a database query promise
+*/
+function createItemsMetadata(req, next) {
```

Comment:
It is a bit confusing to pass in `req` and `next`, since this isn't a direct API method.  You can easily move the `next()` out of the function by moving the `catch()` statement out of the function.  You also do not need the full `req` object, just `req.body`.  You can pass that in the `record` parameter.  For example:
```js
function createItemsMetadata(record) {
  // other methods
  return db.exec(sql, [ record ])
    .then(() => uuid.unparse(record.uuid));
}

// then in the API method
app.post('/inventory', function (req, res, next) {
  core.createItemsMetadata(req.body)
  .then(function (uuid) {
    res.status(201).json({ uuid : uuid });
  })
  .catch(next)
  .done();
});
```